examples/GraphTraversal/Graph.roc

```diff
@@ -0,0 +1,150 @@
+## The Graph interface represents a graph using an adjacency list and exposes
+## functions for working with graphs, such as creating one from a list and
+## performing a depth-first or breadth-first search.
+interface Graph
+    exposes [
+        Graph,
+        fromList,
+        dfs,
+        bfs,
+    ]
+    imports [
+        Dict,
+    ]
+
+## Graph type representing a graph as a dictionary of adjacency lists
+## where each key is a vertex and each value is a list of its adjacent vertices.
+Graph a := Dict a (List a) | a has Eq
+
+## Create a Graph from an adjacency list.
+fromList : List (a, List a) -> Graph a
+fromList = \adjacencyList ->
```

Comment:
Why can't you just use `Dict.fromList`? I guess we need to update it to the new tuple syntax instead of `T a (List a)` for you to use it?

Reply:
Yeah I think that is the issue, worth adding an Issue to track this? 

```

── TYPE MISMATCH ─────────────────────────── examples/GraphTraversal/Graph.roc ─

This 1st argument to fromList has an unexpected type:

21│      Dict.fromList adjacencyList
                       ^^^^^^^^^^^^^

This adjacencyList value is a:

    List (
        a,
        List a,
    )a | a has Hash & Eq, a has Hash & Eq

But fromList needs its 1st argument to be:

    List (T k v) | k has Hash & Eq
```

```coffee
## Create a Graph from an adjacency list.
fromList : List (a, List a) -> Graph a
fromList = \adjacencyList ->
    Dict.fromList adjacencyList
    |> @Graph
```

Reply:
Yeah, should be a good first issue for anyone that wants to fix it. Maybe too simple of an issue though.

Reply:
Done